balsamiq.md

```diff
@@ -0,0 +1,222 @@
+
+It feels more like a documentation site landing page, right? It has a featured articles section with the articles that are most relevant to new users, and the rest of the articles are split evenly in three columns. Yet none of it is hard-coded, even the featured "Getting Started" articles.
+
+I'll start by explaining the "Everything Else..." section at the bottom and what makes it different from our previous version.
+
+The links there are automatically placed by Hugo and styled using the [Bootstrap List group](http://getbootstrap.com/components/#list-group) component. That part is pretty easy. The challenge was putting them in columns and making sure that the columns were equal heights, regardless of how many articles there were.
+
+Here's the code I wrote inside the Hugo template to define some variables to use further down in the code:
+
+```go
+{{ $featuredRows := 1 }} // number of rows to feature in Getting Started section
+{{ $totalCount := len (where .Site.Pages "Section" "desktop") }} // total number of articles in this section
+{{ $rowCount := (sub (div (add (add $totalCount 1) (mod $totalCount 3)) 3) $featuredRows) }} // number of rows in each column
+```
+
+And when it comes time to populate the columns:
+
+```html
+<div class="row mt1">
+  <div class="col-xs-12 col-md-4">
+    <div class="list-group">
+      {{ range first $rowCount (after 3 .Data.Pages.ByWeight) }}
+      <a href="{{ .Permalink }}" class="list-group-item">{{ .Title }}</a>
+      {{ end }}
+    </div>
+  </div>
+  <div class="col-xs-12 col-md-4">
+    <div class="list-group">
+      {{ range first $rowCount (after (add $rowCount 3) .Data.Pages.ByWeight) }}
+      <a href="{{ .Permalink }}" class="list-group-item">{{ .Title }}</a>
+      {{ end }}
+    </div>
+  </div>
+  <div class="col-xs-12 col-md-4">
+    <div class="list-group">
+      {{ range first $rowCount (after (add (mul $rowCount 2) 3) .Data.Pages.ByWeight) }}
+      <a href="{{ .Permalink }}" class="list-group-item">{{ .Title }}</a>
+      {{ end }}
+    </div>
+  </div>
+</div>
+```
+
+The variable called `$featuredRows` helps determine where the count should start for columns. From there Hugo counts the number of remaining articles and divides that number by 3 (rounded up to the nearest whole number). It then creates the number of items in each column so that they are as even as possible. It takes a bit of math to do it, but that's what computers are good at anyway.
+
+And now to that "Getting Started" section at the top...
+
+![](https://media.balsamiq.com/images/docslikecode/desktop-toc-getting-started.png)
+
+I wanted different images for each of the featured articles, but I just didn't like the idea of hard-coding any links or resources, in case we decided to change things later (and also because I'm stubborn). So, the top part of the code identifies the first three articles (ordered by weight), lists their names, links them up, and then grabs an image for each of them using the following code:
+
```

Comment:
Oo, cool. I'm reminded of https://guides.github.com/, where I think they create those plaid/hexagon/etc patterns using a gem, http://jasonlong.github.io/geo_pattern/ has patterns. Your example is a nice use of a filename/template pattern that is more meaningful.